patches/minecraft/net/minecraft/block/BlockRailBase.java.patch

```diff
@@ -39,10 +39,27 @@
 +    }
 +
 +    /**
++     * Return the rail's direction.
++     * Can be used to make the cart think the rail is a different shape,
++     * for example when making diamond junctions or switches.
++     * The cart parameter will often be null unless it it called from EntityMinecart.
++     *
++     * @param world The world.
++     * @param pos Block's position in world
++     * @param state The BlockState
++     * @param cart The cart asking for the metadata, null if it is not called by EntityMinecart.
++     * @return The direction.
++     */
++    public EnumRailDirection getRailDirection(IBlockAccess world, BlockPos pos, IBlockState state, @javax.annotation.Nullable net.minecraft.entity.item.EntityMinecart cart)
++    {
```

Comment:
`@Nullable` is marked as `RetentionPolicy.RUNTIME`, and comes from `com.google.code.findbugs:jsr305`. `jsr305` version `1.3.9` seems to be bundled with the server - it doesn't seem present on the client.